clang/runtime/dpct-rt/include/dpct/memory.hpp

```diff
@@ -1603,6 +1603,23 @@ template <typename T>
 using usm_device_allocator = detail::deprecated::usm_allocator<T, sycl::usm::alloc::shared>;
 } // namespace deprecated
 
+enum class attribute_type {
+  context,
+  memory_type,
+  device_pointer,
+  host_pointer,
+  p2p_tokens,
+  sync_memps,
+  buffer_id,
+  is_managed,
+  device_id,
+  is_legacy_ipc_capable,
+  range_start_addr,
+  range_size,
+  mapped,
+  allowed_handle_types
```

Comment:
I think adding the 1:1 attributes, which we have mapping, is enough.
And add one more "unsupported" enum to map all the unsupported attributes.